test/indexSpec.js

```diff
@@ -471,12 +471,137 @@ describe('PluginBells', function () {
+          }
+        })
+
+        const infoRedLedger = cloneDeep(require('./data/infoRedLedger.json'))
+        const nockInfo = nock('http://blue.example')
+          .get('/')
+          .reply(200, infoRedLedger)
+
+        const nockAccount = nock('http://blue.example')
+          .get('/accounts/mike')
+          .reply(200, {
+            ledger: 'http://blue.example',
+            name: 'mike'
+          })
+
+        plugin.connect().catch((err) => {
+          assert.equal(err.message, 'Unable to set prefix from ledger or from local config')
+          nockInfo.done()
+          nockAccount.done()
+          done()
+        })
+      })
+
+      it('gives an error if ledger and local prefix don\'t match', function (done) {
+        const plugin = new PluginBells({
+          prefix: 'example.red.',
+          account: 'http://blue.example/accounts/mike',
+          password: 'mike',
+          debugReplyNotifications: true,
+          debugAutofund: {
+            connector: 'http://mark.example',
+            admin: {username: 'adminuser', password: 'adminpass'}
+          }
+        })
+
+        const infoRedLedger = Object.assign(
+          cloneDeep(require('./data/infoRedLedger.json')),
+          { ilp_prefix: 'example.blue.' }
+        )
+        const nockInfo = nock('http://blue.example')
+          .get('/')
+          .reply(200, infoRedLedger)
+
+        const nockAccount = nock('http://blue.example')
+          .get('/accounts/mike')
+          .reply(200, {
+            ledger: 'http://blue.example',
+            name: 'mike'
+          })
+
+        plugin.connect().catch((err) => {
+          assert.equal(err.message, 'Unexpected ledger prefix: does not match local config')
+          nockInfo.done()
+          nockAccount.done()
+          done()
+        })
+      })
+
+      it('gets the ledger\'s prefix when available', function * () {
+        const plugin = new PluginBells({
+          // no prefix
+          account: 'http://blue.example/accounts/mike',
+          password: 'mike',
+          debugReplyNotifications: true,
+          debugAutofund: {
+            connector: 'http://mark.example',
+            admin: {username: 'adminuser', password: 'adminpass'}
+          }
+        })
+
+        const infoRedLedger = Object.assign(
+          cloneDeep(require('./data/infoRedLedger.json')),
+          { ilp_prefix: 'example.blue.' }
+        )
+        const nockInfo = nock('http://blue.example')
+          .get('/')
+          .reply(200, infoRedLedger)
+
+        const nockAccount = nock('http://blue.example')
+          .get('/accounts/mike')
+          .reply(200, {
+            ledger: 'http://blue.example',
+            name: 'mike'
+          })
+
+        this.wsRedLedger = new wsHelper.Server('ws://blue.example/accounts/mike/transfers')
+
+        yield plugin.connect()
+        yield assertResolve(plugin.getPrefix(), 'example.blue.')
+        plugin.disconnect()
```

Comment:
disconnect isn't a promise, which is fixed in https://github.com/interledger/js-ilp-plugin-bells/pull/18